### PR TITLE
docs(gpu): mig with k8s

### PR DESCRIPTION
**Pull request by bene2k1**
### Your checklist for this pull request

- [ ] Name your pull request according to the [contribution guidelines](https://github.com/scaleway/docs-content/blob/main/docs/CONTRIBUTING.md).

### Description

MIG with Kubernetes

### CODE REVIEW

File: compute/gpu/how-to/use-mig-with-kubernetes.mdx [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+                        nvidia.com/gpu-driver-upgrade-enabled: true
+    nvidia.com/gpu:     7
+    nvidia.com/gpu:     7
+    nvidia.com/gpu     0           0
+    ```
+
+## Deploy containers that use NVIDIA MIG technology partitions
+
+1. Write a deployment file to deploy 8 pods executing Nvidia SMI. 
+    Open a text editor of your choice and create a deployment file `deploy-mig.yaml`, then paste the following content into the file, save it and exit the editor:
```
[the remark this answers is not in the document]
Comment (bene2k1) [reply compute/gpu/how-to/use-mig-with-kubernetes.mdx:172]: No it is part of the first step

File: compute/gpu/how-to/use-mig-with-kubernetes.mdx [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+<Message type="important">
+  * Scaleway offers GPU Instances such as H100 PCIe GPU Instances, RENDER-S GPU Instance, etc.
+  * Nvidia uses the term *"GPU instance"* to designate a MIG partition of a GPU (MIG= Multi-Instance GPU)
+  * To avoid confusion, we will use the term GPU Instance in this document to refer to the Scaleway GPU Instance, and *"MIG partition"* in the context of the MIG feature.
+</Message>
+
+NVIDIA Multi-Instance GPU (MIG) is a powerful feature that allows you to divide a single NVIDIA GPU into multiple smaller partitions, each with its dedicated GPU resources, such as memory and compute units.
+This technology is particularly valuable in Kubernetes (K8s) environments, where efficient resource allocation is crucial for running diverse workloads efficiently.
+
+In this guide, we will explore capabilities of NVIDIA MIG within a Kubernetes cluster. We cover the steps required to set up and configure MIG-enabled GPUs to use with Kubernetes, to maximize GPU usage and ensure workload isolation and performance predictability.
```
Comment (RoRoJ) [inline compute/gpu/how-to/use-mig-with-kubernetes.mdx:25]:
```suggestion
In this guide, we will explore the capabilities of NVIDIA MIG within a Kubernetes cluster. We cover the steps required to set up and configure MIG-enabled GPUs to use with Kubernetes, to maximize GPU usage and ensure workload isolation and performance predictability.
```

File: compute/gpu/how-to/use-mig-with-kubernetes.mdx [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+    MIG 1g.10gb     Device  1: (UUID: MIG-fdfd2afa-5cbd-5d1d-b1ae-6f0e13cc0ff8)
+    MIG 1g.10gb     Device  2: (UUID: MIG-b2925bc6-41ca-5ccd-bf5e-24259386f88e)
+    MIG 1g.10gb     Device  3: (UUID: MIG-083c76fc-5d21-5322-9d50-c8e21a01852f)
+    MIG 1g.10gb     Device  4: (UUID: MIG-13d8a181-5bc1-5527-9a0f-9c3f9cc1d89e)
+    MIG 1g.10gb     Device  5: (UUID: MIG-db99bb81-dde3-5c95-9778-daa291fce210)
+    MIG 1g.10gb     Device  6: (UUID: MIG-2d636152-57c2-5e73-9654-b1d21d6d21fb)
+    ```
+    Seven MIG `1g.10gb` partitions are available now.
+
+3. Have a look the Nvidia labels on the node (note the label `nvidia.com/mig.config=all-1g.10gb` and `nvidia.com/gpu.product=NVIDIA-H100-PCIe-MIG-1g.10gb`):
```
Comment (RoRoJ) [inline compute/gpu/how-to/use-mig-with-kubernetes.mdx:121]:
```suggestion
3. Look at the Nvidia labels on the node (note the label `nvidia.com/mig.config=all-1g.10gb` and `nvidia.com/gpu.product=NVIDIA-H100-PCIe-MIG-1g.10gb`):
```

File: compute/gpu/how-to/use-mig-with-kubernetes.mdx [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+        image: nvcr.io/nvidia/pytorch:23.07-py3
+        command: [ "nvidia-smi" ]
+        args: ["-L" ]
+        resources:
+        limits:
+            nvidia.com/gpu: 1
+    nodeSelector:
+        nvidia.com/gpu.product : NVIDIA-H100-PCIe-MIG-1g.10gb
+    ```
+
```
Comment (RoRoJ) [inline compute/gpu/how-to/use-mig-with-kubernetes.mdx:327]:
```suggestion
```

File: compute/gpu/how-to/use-mig-with-kubernetes.mdx [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+    pod/test-2 created
+    pod/test-3 created
+    pod/test-4 created
+    pod/test-5 created
+    pod/test-6 created
+    pod/test-7 created
+    pod/test-8 created
+    ```
+
+3. Display the logs of the pods. The pods are printing their UUID with the `nvidia-smi` command:
```
Comment (RoRoJ) [inline compute/gpu/how-to/use-mig-with-kubernetes.mdx:342]:
```suggestion
3. Display the logs of the pods. The pods print their UUID with the `nvidia-smi` command:
```

File: compute/gpu/how-to/use-mig-with-kubernetes.mdx [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+    GPU 0: NVIDIA H100 PCIe (UUID: GPU-717ef73c-2d43-4fdc-76d2-1cddef4863bb)
+    MIG 1g.10gb     Device  0: (UUID: MIG-fdfd2afa-5cbd-5d1d-b1ae-6f0e13cc0ff8)
+    GPU 0: NVIDIA H100 PCIe (UUID: GPU-717ef73c-2d43-4fdc-76d2-1cddef4863bb)
+    MIG 1g.10gb     Device  0: (UUID: MIG-222504cc-4a15-589b-8ec8-dbc02e6fb378)
+    GPU 0: NVIDIA H100 PCIe (UUID: GPU-717ef73c-2d43-4fdc-76d2-1cddef4863bb)
+    MIG 1g.10gb     Device  0: (UUID: MIG-083c76fc-5d21-5322-9d50-c8e21a01852f)
+    GPU 0: NVIDIA H100 PCIe (UUID: GPU-717ef73c-2d43-4fdc-76d2-1cddef4863bb)
+    MIG 1g.10gb     Device  0: (UUID: MIG-fdfd2afa-5cbd-5d1d-b1ae-6f0e13cc0ff8)
+    ```
+    As you can see, seven pods have been executed on different MIG partitions while the eigth pod had to wait for one of the seven MIG partitions to become available in order to be executed.
```
Comment (RoRoJ) [inline compute/gpu/how-to/use-mig-with-kubernetes.mdx:362]:
```suggestion
    As you can see, seven pods have been executed on different MIG partitions, while the eighth pod had to wait for one of the seven MIG partitions to become available in order to be executed.
```

**Review by nerda-codes — APPROVED** [review]

**Review by RoRoJ — APPROVED** [review]

**Pull request merged**